framework/src/materials/MaterialBase.C

```diff
@@ -136,21 +136,29 @@ MaterialBase::initQpStatefulProperties()
 void
 MaterialBase::checkStatefulSanity() const
 {
-  for (const auto & it : _props_to_flags)
-    if (static_cast<int>(it.second) % 2 == 0) // Only Stateful properties declared!
-      mooseError("Material '", name(), "' requests undefined stateful property '", it.first, "'");
+  for (const auto & [id, min_state] : _props_to_min_states)
+    if (min_state > 0)
+      mooseError("The stateful property '",
+                 _fe_problem.getMaterialPropertyRegistry().getName(id),
+                 "' is undefined");
 }
 
 void
-MaterialBase::registerPropName(std::string prop_name, bool is_get, MaterialPropState state)
+MaterialBase::registerPropName(const std::string & prop_name, bool is_get, const unsigned int state)
```

Comment:
Good catch.  We should totally do a `string_view` sweep through MOOSE sometime, though.

Reply:
This one was actually intentional. They modify the string later on. But... I like consistency and this is just an init routine so I don't really care.